### DISCUSSION TITLE
请教大神，我修改了mumu模拟的地址，和adb调试端口为本机的后，为什么不能正常运作

### DISCUSSION THREAD

**Discussion by false111** [lisai9093/YYS#13] [Q&A] [answered]
![微信截图_20240111111213](https://github.com/lisai9093/YYS/assets/122444151/62088fb4-44d5-46bb-9e26-49f1a453f72a)
![微信截图_20240111111237](https://github.com/lisai9093/YYS/assets/122444151/5dc15d27-55eb-4c32-9bde-1d6e4134c685)

**Comment by lisai9093** [comment]
> ![微信截图_20240111111213](https://private-user-images.githubusercontent.com/122444151/295778993-62088fb4-44d5-46bb-9e26-49f1a453f72a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5NDQzNzcsIm5iZiI6MTcwNDk0NDA3NywicGF0aCI6Ii8xMjI0NDQxNTEvMjk1Nzc4OTkzLTYyMDg4ZmI0LTQ0ZDUtNDZiYi05ZTI2LTQ5ZjFhNDUzZjcyYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMVQwMzM0MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZjVlNzNkNjQ5ZWIxOWJhOTk4ZDkzNzc3ZDc2OWJmMGFiNjFhMjZjNTkyMjgyZjY2OTlhMGI3NGY1Nzc5MzlmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.T4a4d2wu6QB_zuecnD6MrHkQo-w4nsNeprSAh65ryvI) ![微信截图_20240111111237](https://private-user-images.githubusercontent.com/122444151/295778999-5dc15d27-55eb-4c32-9bde-1d6e4134c685.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5NDQzNzcsIm5iZiI6MTcwNDk0NDA3NywicGF0aCI6Ii8xMjI0NDQxNTEvMjk1Nzc4OTk5LTVkYzE1ZDI3LTU1ZWItNGMzMi05YmRlLTFkNmU0MTM0YzY4NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMVQwMzM0MzdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YTIxMTU3YjgxNzk5NTNjZTM3OTk0ZjE0ZTRkNzlhODVhMzAwYjhiOTkxZDAwMjJmZDBmZTVmNTcwY2M0ZDM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.yspky1F0r4r0XOVBkz2MfUKbpXl58LA5Zd6cR_L1O3w)

1. 目前只支持一个adb设备，如果有两个会出现错误
2. 错误代码感觉像是screen没有解码成功导致的，你试试用windows自带的终端允许脚本看看

**Reply by false111** [reply]
感谢大神的回复，目前直流一个adb可以用idle启动脚本了，但是我选择探索模式，并不能使用，且一直有个终端运行框一闪一闪的，我就把他停掉，请大神赐教
![微信截图_20240111121717](https://github.com/lisai9093/YYS/assets/122444151/b33b156d-92d0-4586-ac11-83ffa10d883f)

**Reply by lisai9093** [reply]
你试试用17的debug模式能不能看到截图？

**Comment by false111** [comment]
我刚重启了电脑17可以截到图，就是屏幕在那闪那个黑色的终端框，一直闪，好奇怪哦大神

**Reply by lisai9093** [reply]
你用的是终端还是IDLE运行？如果是IDLE应该看不到任何终端才对。有没有截图或者视频？

**Reply by false111** [reply]
我用的是idle运行的，他就是闪终端那个黑框，一闪而过，然后一直闪

**Reply by lisai9093** [reply]
我猜应该是subprocess在运行ADB指令的时候会默认打开新终端窗口并关闭，只是我这里可能速度太快所以没有发现。这个问题我以后可以尝试修复但应该不会影响正常脚本的使用。
如果你运行脚本后没有任何反应（IDLE窗口没有变化），那么应该就是脚本没有检测到对应的界面。比如7是单刷探索，默认只支持困难28，你可以进到探索见面里（有小怪的地方）再试试

**Reply by false111** [reply]
它可以正常使用了，就是屏幕一直闪黑框，其他没问题，是电脑运行速度慢吗

**Reply by false111** [reply]
或者修复的点在哪，我去研究一下大神

**Reply by lisai9093** [reply]
有可能是电脑速度慢，也有可能是安卓模拟器/手机传输慢。我目前找到的隐藏窗口的方法（英文）：https://sukhbinder.wordpress.com/2022/08/07/how-to-hide-the-console-when-using-subprocess/
具体来说就是在运行subprocess.run的时候设置窗口始终为隐藏：
si = subprocess.STARTUPINFO()
si.dwFlags |= subprocess.STARTF_USESHOWWINDOW
subprocess.run(["say", "Check RaspberryPi, its not working"] ,startupinfo=si)

**Reply by lisai9093** [reply]
我等会儿会去测试下，如果没啥大问题的话应该很快就会更新到Github

**Reply by false111** [reply]
谢谢大神，不过你的这个脚本真好用，用了电脑不会卡，应该是我mumu慢，因为我刚发现了探索的时候，有时候会从k28退到庭院

**Reply by lisai9093** [reply]
庭院这个问题解决方法就是把“ditu”这个图片替换成你庭院里的地图就好，因为庭院皮肤不一样所以没办法统一，只能自己手动截图替换。。。

**Reply by false111** [reply]
好滴，我直接用的原始皮，谢谢大神

**Reply by lisai9093** [reply]
刚刚更新完，你再试试还有没有终端闪烁的现象

**Reply by false111** [reply]
谢谢大神，刚用新的试了一下，没有之前那么频繁了，但是还是会有，大概一秒两次

**Reply by false111** [reply]
就是有打印的时候会闪

**Reply by lisai9093** [reply]
我又找了个方法，你更新下再试试看效果怎么样

**Reply by false111** [reply, marked as answer]
谢谢，用idle还是会闪，但是在yys,py文件所在路径用终端运行不会闪

**Reply by lisai9093** [reply]
ADB模式还是推荐用终端比较好，IDLE本身也有不少问题很难修复。。。

**Reply by false111** [reply]
好滴，谢谢大佬，这个点击位置会变化吗大佬，准备挂一晚上哈哈

**Reply by lisai9093** [reply]
位置都是会随机变化的。但是不建议长时间挂机不然还是有鬼使黑风险，一次最好不要超过2小时。
